### ISSUE TITLE
Example SSP Doesn't Use `this-system` Component Approach

### ISSUE THREAD

**Issue by aj-stein-nist** [usnistgov/oscal-content#138]
# User Story:

As an OSCAL tool developer, to understand complete, current examples of system security plans, I want to see this example use the `this-system` component like the other OSCAL SSP examples (for a leveraged and leveraging system, respectively).

## Goals:

- [ ] Update `src/examples/ssp/xml/ssp-example.xml` so the pipeline updates all formats of the OSCAL SSP example.
- [ ] Ensure the example is OSCAL valid.

NOTE: This was originally reported by [@openprivacy via Gitter](https://gitter.im/usnistgov-OSCAL/Lobby?at=63116cf8b16e8236e359202d).

## Dependencies:

N/A

## Acceptance Criteria

- [ ] All readme documentation affected by the changes in this issue have been updated.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

{The items above are general acceptance criteria for all User Stories. Please describe anything else that must be completed for this issue to be considered resolved.}

**Comment by wandmagic** [comment]
I'll take of this over the weekend

**Comment by iMichaela** [comment]
Running the `oscal-cli` to validate the example, it identifies a large set of constraints not being met per OSCAL v1.1.1. It is possible that the example was validated using only the schema which looked only at the formatting but the constraints were not checked. 

[ERROR] [/system-security-plan/system-characteristics[1]/prop[1]/@name] Value 'deployment-model' doesn't match one of 'authenticator-assurance-level, cloud-deployment-model, cloud-service-model, federation-assurance-level, identity-assurance-level, or marking' at path '/system-security-plan/system-characteristics[1]/prop[1]/@name'

      This [ERROR] is a constraint error:
```
      <prop name="deployment-model" value="private" />
```
      should be:
```
       <prop name="cloud-deployment-model" value="private-cloud" />
```
      per OSCAL Reference : [system-characteristics](https://pages.nist.gov/OSCAL-Reference/models/v1.1.1/system-security-plan/xml-reference/#/system-security-plan/system-characteristics)

[ERROR] [/system-security-plan/system-characteristics[1]/prop[2]/@name] Value 'service-models' doesn't match one of 'authenticator-assurance-level, cloud-deployment-model, cloud-service-model, federation-assurance-level, identity-assurance-level, or marking' at path '/system-security-plan/system-characteristics[1]/prop[2]/@name'

      This [ERROR] is a constraint error:
```
      <prop name="service-models" value="iaas" />
```
      should be:
```
      <prop name="cloud-service-models" value="iaas" />
```
      per OSCAL Reference : [system-characteristics](https://pages.nist.gov/OSCAL-Reference/models/v1.1.1/system-security-plan/xml-reference/#/system-security-plan/system-characteristics)

[ERROR] [/system-security-plan/system-implementation[1]/component[2]/prop[2]/@name] Value 'last-modified-date' doesn't match one of 'allows-authenticated-scan, asset-id, asset-tag, asset-type, baseline-configuration-name, function, implementation-point, inherited-uuid, label, leveraged-authorization-uuid, marking, model, network-id, patch-level, public, release-date, sort-id, validation-reference, validation-type, version, virtual, or vlan-id' at path '/system-security-plan/system-implementation[1]/component[2]/prop[2]/@name'

[ERROR] [/system-security-plan/system-implementation[1]/component[3]/prop[1]/@name] Value 'last-modified-date' doesn't match one of 'allows-authenticated-scan, asset-id, asset-tag, asset-type, baseline-configuration-name, function, implementation-point, inherited-uuid, label, leveraged-authorization-uuid, marking, model, network-id, patch-level, public, release-date, sort-id, validation-reference, validation-type, version, virtual, or vlan-id' at path '/system-security-plan/system-implementation[1]/component[3]/prop[1]/@name'
```
line 139:    <prop name="last-modified-date" value="20181015" />
```
should be:
```
                 <prop name="release-date" value="20181015" />
```
OR
```
                 <prop name="last-modified-date" value="20181015" ns="http://csrc.nist.gov/ns/something-else"/>
```

[ERROR] [/system-security-plan/system-implementation[1]/component[4]/prop[1]/@name] Value 'last-modified-date' doesn't match one of 'allows-authenticated-scan, asset-id, asset-tag, asset-type, baseline-configuration-name, function, implementation-point, inherited-uuid, label, leveraged-authorization-uuid, marking, model, network-id, patch-level, public, release-date, sort-id, validation-reference, validation-type, version, virtual, or vlan-id' at path '/system-security-plan/system-implementation[1]/component[4]/prop[1]/@name'

[ERROR] [/system-security-plan/system-implementation[1]/component[5]/prop[1]/@name] Value 'last-modified-date' doesn't match one of 'allows-authenticated-scan, asset-id, asset-tag, asset-type, baseline-configuration-name, function, implementation-point, inherited-uuid, label, leveraged-authorization-uuid, marking, model, network-id, patch-level, public, release-date, sort-id, validation-reference, validation-type, version, virtual, or vlan-id' at path '/system-security-plan/system-implementation[1]/component[5]/prop[1]/@name'

[ERROR] [/system-security-plan/system-implementation[1]/inventory-item[1]/implemented-component[1]] The cardinality '0' is below the required minimum '1' for items matching the expression 'prop[has-oscal-namespace('http://csrc.nist.gov/ns/oscal') and @name='asset-id']'.

[ERROR] [/system-security-plan/system-implementation[1]/inventory-item[1]/implemented-component[2]] The cardinality '0' is below the required minimum '1' for items matching the expression 'prop[has-oscal-namespace('http://csrc.nist.gov/ns/oscal') and @name='asset-id']'.

[ERROR] [/system-security-plan/system-implementation[1]/user[1]/role-id[1]] Key reference not found in index 'index-metadata-role-id' for item at path '/system-security-plan/system-implementation[1]/user[1]/role-id[1]'

```
line 81:          <role-id>asset-administrator</role-id>
```
The role was not defined in the metadata section . See [role-id](https://pages.nist.gov/OSCAL-Reference/models/v1.1.1/system-security-plan/xml-reference/#/system-security-plan/system-implementation/user/role-id)

[ERROR] [/system-security-plan/system-implementation[1]/user[2]/role-id[1]] Key reference not found in index 'index-metadata-role-id' for item at path '/system-security-plan/system-implementation[1]/user[2]/role-id[1]'

[ERROR] [/system-security-plan/system-implementation[1]/inventory-item[1]] Key reference not found in index 'index-metadata-role-id' for item at path '/system-security-plan/system-implementation[1]/inventory-item[1]/responsible-party[1]'

[ERROR] [/system-security-plan/system-implementation[1]/inventory-item[1]] Key reference not found in index 'index-metadata-role-id' for item at path '/system-security-plan/system-implementation[1]/inventory-item[1]/responsible-party[2]'

[ERROR] [/system-security-plan/system-implementation[1]/inventory-item[1]/responsible-party[1]] Key reference not found in index 'index-metadata-role-id' for item at path '/system-security-plan/system-implementation[1]/inventory-item[1]/responsible-party[1]'

[ERROR] [/system-security-plan/system-implementation[1]/inventory-item[1]/responsible-party[2]] Key reference not found in index 'index-metadata-role-id' for item at path '/system-security-plan/system-implementation[1]/inventory-item[1]/responsible-party[2]'


@wandmagic  - Thank you for addressing this issue.  Hopefully the guidance above helps. Best of luck.

**Comment by iMichaela** [comment]
Addressed by PR #242
